commit ece30470f22c624c29829e5f90d5fea316edb1cf
Author: Mark Robert Miller <markrmiller@apache.org>
Date:   Sat Jan 28 02:19:42 2012 +0000

    try making sure we wait for any recovery to stop for each core in core container shutdown
    
    git-svn-id: https://svn.apache.org/repos/asf/lucene/dev/trunk@1236959 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/solr/core/src/java/org/apache/solr/core/CoreContainer.java b/solr/core/src/java/org/apache/solr/core/CoreContainer.java
index e55ffb7..2fada09 100644
--- a/solr/core/src/java/org/apache/solr/core/CoreContainer.java
+++ b/solr/core/src/java/org/apache/solr/core/CoreContainer.java
@@ -434,6 +434,8 @@ public class CoreContainer
         for (SolrCore core : cores.values()) {
           try {
              core.close();
+             // make sure we wait for any recoveries to stop
+             core.getUpdateHandler().getSolrCoreState().cancelRecovery();
           } catch (Throwable t) {
             SolrException.log(log, "Error shutting down core", t);
           }
